modules/saml/lib/Auth/Source/SP.php

```diff
@@ -239,35 +248,37 @@ private function startSSO2(SimpleSAML_Configuration $idpMetadata, array $state)
 			$ar->setNameIdPolicy($policy);
 		}
 
-		if (isset($state['saml:IDPList'])) {
-			$IDPList = $state['saml:IDPList'];
-		} else {
-			$IDPList = array();
-		}
-		
-		$ar->setIDPList(array_unique(array_merge($this->metadata->getArray('IDPList', array()), 
-												$idpMetadata->getArray('IDPList', array()),
-												(array) $IDPList)));
-		
-		if (isset($state['saml:ProxyCount']) && $state['saml:ProxyCount'] !== null) {
-			$ar->setProxyCount($state['saml:ProxyCount']);
-		} elseif ($idpMetadata->getInteger('ProxyCount', null) !== null) {
-			$ar->setProxyCount($idpMetadata->getInteger('ProxyCount', null));
-		} elseif ($this->metadata->getInteger('ProxyCount', null) !== null) {
-			$ar->setProxyCount($this->metadata->getInteger('ProxyCount', null));
-		}
-		
+		$IDPList = array();
 		$requesterID = array();
-		if (isset($state['saml:RequesterID'])) {
-			$requesterID = $state['saml:RequesterID'];
-		}
-		
-		if (isset($state['core:SP'])) {
-			$requesterID[] = $state['core:SP'];
+
+		if (!$this->disableScoping && !$idpMetadata->getBoolean('disable.scoping', NULL)) {
+			if (isset($state['saml:IDPList'])) {
+				$IDPList = $state['saml:IDPList'];
+			}
+
+			if (isset($state['saml:ProxyCount']) && $state['saml:ProxyCount'] !== null) {
+				$ar->setProxyCount($state['saml:ProxyCount']);
+			} elseif ($idpMetadata->getInteger('ProxyCount', null) !== null) {
+				$ar->setProxyCount($idpMetadata->getInteger('ProxyCount', null));
+			} elseif ($this->metadata->getInteger('ProxyCount', null) !== null) {
+				$ar->setProxyCount($this->metadata->getInteger('ProxyCount', null));
+			}
+
+			if (isset($state['saml:RequesterID'])) {
+				$requesterID = $state['saml:RequesterID'];
+			}
+
+			if (isset($state['core:SP'])) {
+				$requesterID[] = $state['core:SP'];
+			}
 		}
-		
+
+		$ar->setIDPList(array_unique(array_merge($this->metadata->getArray('IDPList', array()),
+				$idpMetadata->getArray('IDPList', array()),
+				(array) $IDPList)));
```

Comment:
Should this not be inside the if-statement? Because if scoping is disabled, we'd never want to setIDPList?